sites/game-client/input-manager/input-manager.mjs

```diff
@@ -319,37 +346,83 @@ class InputManager {
 class JoyStick {
   /**
    * @constructor
-   * @param {Element} containerDiv - The container of the joystick.
+   * @param {InputManager} inputManager - The input manager which this joystick belongs to.
+   * @param {Element} containerDiv - The outer container of the joystick.
+   * @param {Element} controlPointContainerDiv - The container of the joystick control point.
    * @param {Element} controlPointDiv - The div to display the stick of the joystick.
+   * @param {Element} ghostModeDiv - The div to switch on/off ghost mode.
+   * @param {Element} gameButtonDiv - The div used by game (usually by extensions).
    */
-  constructor(containerDiv, controlPointDiv) {
+  constructor(inputManager, containerDiv, controlPointContainerDiv, controlPointDiv, ghostModeDiv, gameButtonDiv) {
```

Comment:
Not really sure if we should hard code the list of buttons we've, but if making it flexible takes too much time, then we can consider adding a TODO here.

Reply:
Answered above.